main/webservices/api/test_api.php

```diff
@@ -2,116 +2,531 @@
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            null, null
+        );
+
+        // move the session to a non-standard URL
+        if (api_is_multiple_url_enabled()) {
+            $urlId = UrlManager::add('https://www.url.org/chamilo-lms/' . time(), 'Non-default URL', 1);
+            UrlManager::update_urls_rel_session([$modelSessionId], $urlId);
+        }
+
+        // create an extra field and set its value in the model session - the new session will be given a different value in this field
+        $extraFieldModel = new ExtraField('session');
+        $EXTRA_FIELD_NAME = 'extraField'.time();
+        $EXTRA_FIELD_VALUE_FOR_MODEL_SESSION = 'extra field value for model';
+        $EXTRA_FIELD_VALUE_FOR_NEW_SESSION = 'extra field value for new session';
+        $extraFieldId = $extraFieldModel->save([
+            'field_type' => ExtraField::FIELD_TYPE_TEXT,
+            'variable' => $EXTRA_FIELD_NAME,
+            'display_text' => $EXTRA_FIELD_NAME,
+            'visible_to_self' => 1,
+            'visible_to_others' => 1,
+            'changeable' => 1,
+            'filter' => 1,
+        ]);
+        SessionManager::update_session_extra_field_value($modelSessionId, $EXTRA_FIELD_NAME, $EXTRA_FIELD_VALUE_FOR_MODEL_SESSION);
+
+        // create a second extra field and set its value in the model session - the new session will inherit the same value in this field
+        $SECOND_EXTRA_FIELD_NAME = 'secondExtraField'.time();
+        $SECOND_EXTRA_FIELD_VALUE = 'second extra field value';
+        $secondExtraFieldId = $extraFieldModel->save([
+            'field_type' => ExtraField::FIELD_TYPE_TEXT,
+            'variable' => $SECOND_EXTRA_FIELD_NAME,
+            'display_text' => $SECOND_EXTRA_FIELD_NAME,
+            'visible_to_self' => 1,
+            'visible_to_others' => 1,
+            'changeable' => 1,
+            'filter' => 1,
+        ]);
+        SessionManager::update_session_extra_field_value($modelSessionId, $SECOND_EXTRA_FIELD_NAME, $SECOND_EXTRA_FIELD_VALUE);
+
+        // subscribe the model session to the promotion - the new session will inherit this too
+        SessionManager::subscribe_sessions_to_promotion($promotionId, [$modelSessionId]);
+
+        // create courses and add them the the model session
+        $COURSE_CODES = [ 'course A'.time(), 'course B'.time(), 'course C'.time() ];
+        $courseList = [];
+        foreach($COURSE_CODES as $code) {
+            $course = CourseManager::create_course(['code' => $code, 'title' => $code, 'wanted_code' => $code ], UserManager::get_user_id_from_username(self::WEBSERVICE_USERNAME));
+            $courseList[] = $course['real_id'];
+        }
+        SessionManager::add_courses_to_session($modelSessionId, $courseList);
+
+        // call the webservice to create the new session from the model session, specifying a different value in the first extra field
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 134 characters](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)